File: sitoEratostenesa/sieve.cpp
// g++ 7.2.0
#include <cstdio>
#include <cmath>
#include <vector>

bool isPrime(int n) {
  if(n<2) {
    return false;
  }
  for(int i=2; i<=sqrt(n); i++) {
    if(!(n%i)) {
      return false;
    }
  }
  return true;
}

std::vector <int> sieve(int n) {
  if(n>0) {
    std::vector <int> primes;
    int *array = new int[n];
    for(int i = 1; i<(n+1); i++) {
      array[i] = i;
    }
    for(int i = 0; i<n; i++) {
      if(isPrime(array[i])) {
        primes.push_back(array[i]);
        for(int j=2*i; j<n; j+=i) {
          array[j] = 0;
        }
      }
    }
    delete[] array;
    return primes;
  }
}

int main() {
  std::vector <int> primes = sieve(100000000);
  for(std::vector<int>::iterator i = primes.begin(); i != primes.end(); i++) {
    printf("%d\n", *i);
  }
}

File: sitoEratostenesa/sieve.py
# Python 3.6.2
import math

def isPrime(n):
    if n<2:
        return False
    return all(n%i for i in range(2, int(math.sqrt(n))+1))

def sieve(n):
    array = list(range(0, n+1))
    for i in range(0, n+1):
        if isPrime(array[i]):
            for j in range(i*2, n)[::i]:
                array[j] = 0
        else:
            array[i] = 0
    return list(filter(lambda x: x!=0, array))

print(sieve(2137000))
